src/test/e2e/37_component_status_test.go

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package test provides e2e tests for Zarf.
+package test
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/zarf-dev/zarf/src/types"
+)
+
+func TestComponentStatus(t *testing.T) {
+	t.Log("E2E: Component Status")
+	tmpDir := t.TempDir()
+	_, _, err := e2e.Zarf(t, "package", "create", "src/test/packages/37-component-status", "-o", tmpDir, "--confirm")
+	require.NoError(t, err)
+	packageName := fmt.Sprintf("zarf-package-component-status-%s.tar.zst", e2e.Arch)
+	path := filepath.Join(tmpDir, packageName)
+	// Stop channel getting the zarf state
+	stop := make(chan bool)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		deployingSeen := false
+		for {
+			select {
+			case <-stop:
```

Comment:
For sake of test resilience - would it make sense to have a timeout case as well?